### PR TITLE
Refactor build.sh script

### PR DESCRIPTION
**Pull request by nghiant2710**
- Add `build` and `list` commands (it was `--list` option).
- Add build command arguments (--device, --os-version, --src, --dest-dir)
- Add support for list of balenaOS versions (as space separated list argument)

Change-type: minor
Signed-off-by: Trong Nghia Nguyen <nghiant2710@gmail.com>

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -4,7 +4,7 @@ RUN apt-get update && apt-get install -y curl wget build-essential libelf-dev aw
 COPY . /usr/src/app
 WORKDIR /usr/src/app
 
-ENV VERSION '2.29.0+rev1.prod'
-RUN ./build.sh %%BALENA_MACHINE_NAME%% $VERSION example_module
+ENV VERSION '2.29.0+rev1.prod 2.48.0+rev3.prod'
+RUN ./build.sh --device %%BALENA_MACHINE_NAME%% --os-version $VERSION --src example_module
```
Comment (shaunmulligan) [inline Dockerfile.template]: i think this needs to be `RUN ./build.sh build ...`

File: build.sh [hunk 2]
```diff
@@ -147,28 +164,66 @@ function get_and_build()
...
+		{
+			echo "error: unknown command: $1"
+			usage
+		} >&2
+		exit 1
+		;;
+esac
 
 [[ -d "$module_dir" ]] || fatal "ERROR: Cannot find module directory $module_dir"
+[[ -z "$device" ]] && fatal "ERROR: No device specified"
```
Comment (shaunmulligan) [inline build.sh]: If no device type is specified we should use `%%BALENA_MACHINE_NAME%%` but default and just print a warning that no device type was specified so using the default.

File: Dockerfile.template (hunk 1 above)
Comment (nghiant2710) [reply Dockerfile.template]: Fixed

File: build.sh (hunk 2 above)
Comment (nghiant2710) [reply build.sh]: Done

**Comment by nghiant2710** [comment]
hey @shaunmulligan, I fixed your comments, can you please check this PR again?

File: build.sh [hunk 3]
```diff
@@ -147,28 +164,77 @@ function get_and_build()
...
+		--dest-dir) output_dir="$1" && shift ;;
+        --src) module_dir="$1" && shift ;;
+        --) break ;;
+        *)
+            {
+                echo "error: unknown flag: $flag"
+                usage
+            } >&2
+            exit 1
+            ;;
```
Comment (floion) [inline build.sh:191]: Guess it needs same spacing for all lines

Comment (nghiant2710) [reply build.sh:191]: @floion, it looks fine to me, please correct me if i'm wrong.

**Review by shaunmulligan — APPROVED** [review]
@nghiant2710 this looks good to me

**Pull request merged**